exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/ProfileIterator.java

```
@@ -0,0 +1,216 @@
+
+import javax.annotation.Nullable;
+import java.sql.Timestamp;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * DRILL-5068: Add a new system table for completed profiles
+ */
+public class ProfileIterator implements Iterator<Object> {
+  static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(ProfileIterator.class);
+
+  private final QueryProfileStoreContext profileStoreContext;
+  private final Iterator<?> itr;
+  private final String queryingUsername;
+  private final boolean isAdmin;
+
+  public ProfileIterator(FragmentContext context) {
+    //Grab profile Store Context
+    profileStoreContext = context
+        .getDrillbitContext()
+        .getProfileStoreContext();
+
+    queryingUsername = context.getQueryUserName();
+    isAdmin = hasAdminPrivileges(context.getQueryContext());
+
+    itr = iterateProfileInfo(context);
+  }
+
+  private boolean hasAdminPrivileges(QueryContext queryContext) {
+    if (queryContext == null) {
+      return false;
+    }
+    QueryOptionManager options = queryContext.getOptions();
+    if (queryContext.isUserAuthenticationEnabled() &&
+        !ImpersonationUtil.hasAdminPrivileges(
+          queryContext.getQueryUserName(),
+          ExecConstants.ADMIN_USERS_VALIDATOR.getAdminUsers(options),
+          ExecConstants.ADMIN_USER_GROUPS_VALIDATOR.getAdminUserGroups(options))) {
+      return false;
+    }
+
+    //Passed checks
+    return true;
+  }
+
+  private Iterator<ProfileInfo> iterateProfileInfo(FragmentContext context) {
+    try {
+      PersistentStore<UserBitShared.QueryProfile> profiles = profileStoreContext.getCompletedProfileStore();
+
+      return transform(getAuthorizedProfiles(profiles.getAll(), queryingUsername, isAdmin));
+
+    } catch (Exception e) {
+      logger.error(String.format("Unable to get persistence store: %s, ", profileStoreContext.getProfileStoreConfig().getName(), e));
```

`Exception e` should be an argument to `logger.error` not `String.format`

---

I was trying to format the error message to indicate the issue., but I realize I might be hiding the actual error message. +1